LocalPackages/SyncUI/Sources/SyncUI/Views/SyncSettingsViewExtension.swift

```diff
@@ -97,6 +107,18 @@ extension SyncSettingsView {
                 })
             })
             .disabled(!model.isAccountRecoveryAvailable)
+            .alert(isPresented: alertBinding) {
+                Alert(
+                    title: Text("Secure Your Device to Use Sync & Backup"),
+                    message: Text("A device password is required to use Sync & Backup."),
+                    dismissButton: .default(Text("Go to Settings"), action: {
+                        UIApplication.shared.open(URL(string: UIApplication.openSettingsURLString)!,
+                                                  options: [:],
+                                                  completionHandler: nil)
+                        model.shouldShowPasscodeRequiredAlert = false
+                    })
+                )
```

Comment:
This alert is not presented when Sync is enabled. To have it show up also when "Sync With Another Device" under devices list is tapped, this alert should be defined in SyncSettingsView as a modifier on the top-level List (around the line 91 in SyncSettingsView.swift).

Additionally, the dialog is not shown when tapping "Save Recovery PDF" when Sync is enabled, because that action calls delegate directly instead of wrapping it in `commonAuthenticate()`:

```swift
func saveRecoveryPDF() {
    delegate?.shareRecoveryPDF()
}
```

This is probably because the delegate (the VC) performs authentication on its own, but it doesn't report failure back to the view model.